hypervisor/src/mshv/mod.rs

```diff
@@ -562,6 +562,18 @@ impl cpu::Vcpu for MshvVcpu {
                     debug!("Exception Info {:?}", { info.exception_vector });
                     Ok(cpu::VmExit::Ignore)
                 }
+                hv_message_type_HVMSG_X64_APIC_EOI => {
+                    let info = x.to_apic_eoi_info().unwrap();
+                    // The kernel should dispatch the EOI to the correct thread.
+                    // Check the VP index is the same as the one we have.
+                    assert!(info.vp_index == self.vp_index as u32);
```

Comment:
Is it possible for EOI to be delivered to a different VP (than the one where the interrupt was received)? If such a thing does happen, does it mean the hypervisor is faulty?

Reply:
Any VP running inside the guest can generate such an event. Hypervisor doesn't really shuffle things from one VP to another. Like I said in the comment, if the VP index is wrong then there is a bug in our kernel driver.

Reply:
Thanks for clarifying.